File: ekart-backend/Order-Service/src/main/java/com/ekart/order/service/WishListService.java
package com.ekart.order.service;

import java.util.List;

import com.ekart.order.entity.Wishlist;

public interface WishListService {

	public String addItems(Wishlist wishlist);

	public List<Wishlist> getAllItemsInWishList();

	public Wishlist getItemByProdId(int prodId);

	public List<Wishlist> getItemByEmail(String email);

	public Wishlist getItemByProdIdAndType(int prodId, String inventoryType);

	public String deleteItemById(int wishlistId);

}

File: ui/ui-frontend/src/Services/WishListService.js
import { authenticateApi } from "./axiosConfig";

export const WishListService = {
    async getWishListByUser(email){
        const res = await authenticateApi.request({
            url: `/api/wishlist/getItemByEmail/${email}`,
            method: "GET",
          });
          return res;
    },

    async addItemsToWishList(wishList) {
        const res = await authenticateApi.request({
          url: `/api/wishlist/addToWishList`,
          method: "POST",
          data: wishList,
        });
        return res;
    },

    async getItemByTypeAndId(inventoryType,prodId){
        const res = await authenticateApi.request({
            url:`/api/wishlist/getItemByProdIdAndInventoryType/${inventoryType}/${prodId}`,
            method:"GET"
        });
        return res
    },

    async deleteItemFromWishList(wishListId){
        const res = await authenticateApi.request({
            url:`/api/wishlist/deleteItemById/${wishListId}`,
            method:"DELETE"
        })
        return res
    }

    


}
